sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -116,7 +116,7 @@ private[hive] class SparkExecuteStatementOperation(
     val offset = iter.getPosition
     val rows = iter.take(maxRows).toList
```

Comment:
I think method coplexity should not be dependent on the input stracture and the current situation prove it

Reply:
These are private methods, and we can simply require `toRowBasedSet` to take a `IndexedSeq` instead of `Seq` as proposed by @pan3793, right ?
I would have actually preferred this - as it minimizes the change.